cmd/fs-v1.go

```diff
@@ -422,9 +424,18 @@ func (fs *FSObjects) DeleteBucket(ctx context.Context, bucket string) error {
 		return toObjectErr(err, bucket)
 	}
 
-	// Attempt to delete regular bucket.
-	if err = fsRemoveDir(ctx, bucketDir); err != nil {
-		return toObjectErr(err, bucket)
+	if !forceDelete {
+		// Attempt to delete regular bucket.
+		if err = fsRemoveDir(ctx, bucketDir); err != nil {
+			return toObjectErr(err, bucket)
+		}
+	} else {
+		tmpBucketPath := pathJoin(fs.fsPath, minioMetaTmpBucket, fs.fsUUID, bucket)
+		if err = fsSimpleRenameFile(ctx, bucketDir, tmpBucketPath); err != nil {
+			return toObjectErr(err, bucket)
+		}
+
+		fsRemoveAll(ctx, tmpBucketPath) // ignore returned error if any.
```

Comment:
Can we not do the actual remove asnychronously (separate go routine)?
That way we can return immediately since it is moved to the temp directory anyways ...

Reply:
done